src/python/WMCore/HTTPFrontEnd/RequestManager/ReqMgrRESTModel.py

```diff
@@ -272,7 +272,10 @@ def getRequest(self, requestName=None):
         if requestName == None:
             result = GetRequest.getRequests()
         else:
-            result = Utilities.requestDetails(requestName)
+            try:
+                result = Utilities.requestDetails(requestName)
+            except: # if we fail to get a request return nothing
```

Comment:
Seangchan,
The exception I got is HTTPERror 404, 'Given requestName not found'.
May be I should be explicit and catch HTTPError exception only.
But the point is that requestDetails call can fail for whatever reason and
since nobody cares in this call how to handle it the entire stack will fail
elsewhere. If you know better behavior for this let me know and I can update
PR.
V.

On  0, ticoann <notifications@github.com> wrote:
> > @@ -272,7 +272,10 @@ def getRequest(self, requestName=None):
> >          if requestName == None:
> >              result = GetRequest.getRequests()
> >          else:
> > -            result = Utilities.requestDetails(requestName)
> > +            try:
> > +                result = Utilities.requestDetails(requestName)
> > +            except: # if we fail to get a request return nothing
> 
> Valentin, why this exception occurs?
> Is it due to passing the argument in request  body?
> 
> Earlier message you send through email was following. 
> ```
> Once I fixed that like this, I replaced
> 
> result = Utilities.requestDetails(requestName)
> 
> with
> 
>            try:
>                result = Utilities.requestDetails(requestName)
>                print("### getRequest, result", result)
>            except Exception as exp:
>                print("### getRequest, exp", str(exp))
>                return
> 
> my PUT request proceeds and I got running unittest. It fails now in another place
> 
> DEBUG:cherrypy.error:call to PUT with args: ['message', 'Taizong_160106_101656_8214'] kwargs: {} resulted in cannot parse JSON description: error
> 
>                                              stack trace: Traceback (most recent call last):
>  File "/mnt/disk1/vk/WMCoreTests/unittestdeploy/wmagent/1.0.8.pre7/sw.pre/slc6_amd64_gcc481/cms/wmagent/1.0.8.pre7/lib/python2.6/site-packages/WMCore/WebTools/RESTApi.py", line 103, in _buildResponse
>    args, kwargs)
>  File "/mnt/disk1/vk/WMCoreTests/unittestdeploy/wmagent/1.0.8.pre7/sw.pre/slc6_amd64_gcc481/cms/wmagent/1.0.8.pre7/lib/python2.6/site-packages/WMCore/WebTools/RESTModel.py", line 117, in handler
>    data = methodCall(*params, **kwargs)
>  File "/mnt/disk1/vk/WMCoreTests/unittestdeploy/wmagent/1.0.8.pre7/sw.pre/slc6_amd64_gcc481/cms/wmagent/1.0.8.pre7/lib/python2.6/site-packages/WMCore/WebTools/RESTModel.py", line 192, in wrapper
>    return func(**input_data)
>  File "/mnt/disk1/vk/WMCoreTests/unittestdeploy/wmagent/1.0.8.pre7/sw.pre/slc6_amd64_gcc481/cms/wmagent/1.0.8.pre7/lib/python2.6/site-packages/WMCore/HTTPFrontEnd/RequestManager/ReqMgrRESTModel.py", line 629, in putMessage
>    message = JsonWrapper.loads( cherrypy.request.body.read() )
>  File "/mnt/disk1/vk/WMCoreTests/unittestdeploy/wmagent/1.0.8.pre7/sw.pre/slc6_amd64_gcc481/cms/wmagent/1.0.8.pre7/lib/python2.6/site-packages/WMCore/Wrappers/JsonWrapper/__init__.py", line 40, in loads
>    raise ex
> DecodeError: cannot parse JSON description: error
> ```
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/dmwm/WMCore/pull/6495/files#r49096771